packages/shared-core/src/filters.ts

```diff
@@ -423,14 +552,56 @@ export const buildQuery = (filter: SearchFilter[]) => {
       }
     }
   })
-
   return query
 }
 
+export const buildQuery = (
+  filter?: SearchFilterGroup | SearchFilter[]
+): SearchFilters | undefined => {
```

Comment:
This one is called `buildQuery` but actually is just migrating the old filter structure into the new structure I think? Maybe I'm wrong, but it's just difficult with the naming here to understand the purpose of each function here, so either a naming update or more comments/descriptions would really help.

Reply:
The `buildQuery` function takes our client side filter definitions and converts them into a grouped search query every time the filters are run. The `processSearchFilters` will only step in and migrate if there happens to be legacy config stored in their components or other areas of their app.

I do agree though that it the interchangeable use of `filter` and `query` throughout Budibase is very confusing. I will add more commenting here to outline what's going on 👍